src/components/Forecast/index.js

```diff
@@ -16,7 +22,14 @@ class Forecast extends React.Component {
   }
 
   render() {
-    const { isFetching, location, forecastData } = this.props;
+    const { isFetching, location, forecastData, errorMessage } = this.props;
+    const hasError = () => (errorMessage && !forecastData.length);
```

Comment:
Extract outside class and give `errorMessage` and `forecastData` as arguments. Otherwise it can be simply variable.
Also you can achieve this in `mapStateToProps` by:

```javascript
const checkErrors = ({ forecastData, errorMessage }) => ({
  errorMessage,
  forecastData,
  hasError: (errorMessage && !forecastData.length),
})

const mapStateToProps = (state) => ({
  ...checkErrors({ forecastData: getForecastData(state), errorMessage: getErrorMessage(state) }),
})
```

This way you are removing data operation from presentation layer.

Reply:
Nice, thanks :)